library/components/TColumnConfig.vue

```diff
@@ -0,0 +1,21 @@
+<script>
+export default {
+  name: "TColumnConfig",
+  props: {
+    columnConfig: {
+      type: Array,
+      default() {
+        return {};
+      },
+    },
+    setColumnConfig: {
+      type: Function,
+    },
+  },
+  created() {
+    if (typeof this.setColumnConfig === "function") {
+      this.setColumnConfig(this.columnConfig);
+    }
+  },
```

Comment:
Oh absolutely, this whole thing is a super hacky way to avoid having to manually manipulate the template such that the column configs end up in the parent table. It was just the least hacky approach I could think of. There was a reason events weren't working, but I can't remember it now. /shrug

And no worries about not running it, I just got it working in expandable modules.